### PR TITLE
Bruce/24 datasets list

### PR DESCRIPTION
**Pull request by BruceRodrigues**

### CODE REVIEW

File: next/components/Network/components/Datasets/components/MainColumn/projectsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  projects: ProjectsEntityResponse[];
+}
+
+export const ProjectsTable = ({
+  networkPath,
+  projects,
+}: ProjectsTableProps): JSX.Element => {
+  return (
+    <DetailViewTable
+      columns={getProjectsTableColumns(networkPath) as any}
```
Comment (BruceRodrigues) [inline next/components/Network/components/Datasets/components/MainColumn/projectsTable.tsx]: I'm casting this to `any` just so typescript stop complaining about some missing property. Needs more investigation to remove the cast.

File: next/@types/network.ts [hunk 2]
```diff
@@ -29,24 +30,33 @@ export interface Atlas {
...
   coordinators: Coordinator[];
+  datasetQueryOrgans: DatasetQueryOrgan[];
   key: NetworkKey;
   name: string;
   path: string;
 }
 
 export type NetworkParam = {
   network: Network;
+  projects: ProjectsEntityResponse[];
```
Comment (frano-m) [inline next/@types/network.ts]: Hi @BruceRodrigues, we may need to use an aggregated response for example `disease` from `AggregatedDonorOrganismsResponse` aka `donorOrganisms`.

I think change all uses of `ProjectsEntityResponse` to `ProjectsResponse` here and elsewhere, and update the property on NetworkParam and also the NetworkContext to something like `projectsResponses` so there is no confusion over the `projects` property in `ProjectsEntityResponse`.

File: next/components/Network/components/Datasets/components/MainColumn/projectsTable.tsx (hunk 1 above)
Comment (frano-m) [reply next/components/Network/components/Datasets/components/MainColumn/projectsTable.tsx]: The casting can be resolved by adding:

`as typeof DXDetailViewTable` to https://github.com/clevercanary/hca-data-portal/blob/4288d45d4331c95912550e0b43b4ca7c7038f66f/next/components/common/Table/table.styles.ts to the DetailViewTable styled component.

File: next/viewModelBuilders/viewModelBuilders.ts [hunk 3]
```diff
@@ -71,3 +107,98 @@ function getAtlasesTissueColumnDef(): ColumnDef<Atlas> {
...
+function getProjectsCellCountColumnDef(): ColumnDef<ProjectsEntityResponse> {
+  return {
+    accessorKey: "count",
+    cell: "Project-TBD",
+    header: "Cell Count Est.",
+  };
+}
+
+/**
+ * Returns projecst species column def.
```
Comment (frano-m) [inline next/viewModelBuilders/viewModelBuilders.ts]: Update `projecst` to `projects` here and elsewhere 😄

**Review by frano-m — CHANGES_REQUESTED** [review]
Hey @BruceRodrigues just a few feedback items and this one is good to go!

⭐

File: next/utils/networkPages.ts [hunk 4]
```diff
@@ -12,9 +15,23 @@ export const getStaticPaths: GetStaticPaths = async () => {
...
+  const projectsResponses = [];
 
-  return { props: { network } };
+  if (network.datasetQueryOrgans.length > 0) {
+    const result = await fetchEntitiesFromQuery(
+      `${url}/projects`,
+      filterSpecimenOrgan(network.datasetQueryOrgans),
+      undefined
+    );
+    const projects = result.hits.map((item) => item.projects[0]);
```
Comment (frano-m) [inline next/utils/networkPages.ts]: Hi @BruceRodrigues would you mind updating the `projectsResponses` type to `ProjectsResponse` from [here](https://github.com/clevercanary/hca-data-portal/blob/875e072d787043cb064666309245369c629f23bc/next/apis/azul/hca-dcp/common/responses.ts).  Update here, and in `NetworkParam` and elsewhere! Currently it is `ProjectResponse[]`. We may need `disease` from `donorOrganisms` from `AggregatedDonorOrganismsResponse` [see here](https://github.com/clevercanary/data-browser/blob/001836735efd803a1e46d02ac432df56e9fb2aef/explorer/app/apis/azul/hca-dcp/common/responses.ts).
Update line `32` to return the `result.hits`.
Thank you!

**Review by frano-m — CHANGES_REQUESTED** [review]
Hi @BruceRodrigues, one more change request - thank you!

**Review by frano-m — APPROVED** [review]

**Pull request merged**